### PR TITLE
feat(docs): clarify regular expression handling for match artifacts

### PR DESCRIPTION
**Pull request by maggieneterval**
Related to https://github.com/spinnaker/spinnaker/issues/3011, https://github.com/spinnaker/spinnaker/issues/5233

Clarifies regular expression handling of match artifact values. Will be referenced by Deck's Match Artifact tooltip: https://github.com/spinnaker/deck/pull/7792

### CODE REVIEW

File: reference/artifacts/in-pipelines/index.md [hunk 1]
```diff
@@ -38,12 +38,19 @@ artifact.
 The matching behavior is as follows: The artifact to match declared by the
 expected artifact has the same format as a regular artifact; however, its
 values are interpreted as regular expressions to match the corresponding values
-in the incoming artifact.
+in the incoming artifact:
+- Normal strings must match exactly for the artifact to match because Spinnaker
+effectively treats them as regular expressions. For example, Spinnaker
+interprets `inputstring` as `^inputstring$`.
+- Explicit regular expressions are also valid. For example, Spinnaker will
```
Comment (dorbin) [inline reference/artifacts/in-pipelines/index.md]:
```suggestion
- Explicit regular expressions are also valid. For example, Spinnaker
```

File: reference/artifacts/in-pipelines/index.md [hunk 2]
```diff
@@ -38,12 +38,19 @@ artifact.
...
+ match `.*inputstring.*` to any string including the substring `inputstring`.
```
Comment (dorbin) [inline reference/artifacts/in-pipelines/index.md]:
```suggestion
 matches `.*inputstring.*` to any string including the substring `inputstring`.
```

**Review by dorbin — APPROVED** [review]

**Review by ezimanyi — APPROVED** [review]
🍳 :butterfly: :expressionless: :amphora: :heavy_check_mark:

File: reference/artifacts/in-pipelines/index.md (hunk 1 above)
Comment (maggieneterval) [reply reference/artifacts/in-pipelines/index.md]: Great call, thanks for the speedy review!

**Pull request merged**